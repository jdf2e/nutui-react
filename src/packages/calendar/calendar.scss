@import '../popup/popup.scss';
.nut-calendar {
  position: relative;
  display: flex;
  flex: 1;
  color: $calendar-base-color;
  font-size: $calendar-base-font;
  background-color: $white;
  overflow: hidden;
  height: 100%;
  flex-direction: column;

  &.nut-calendar-tile {
    .nut-calendar-header {
      .calendar-title {
        font-size: $calendar-base-font;
      }
    }
  }

  .nut-calendar-taro {
    height: 60vh;
  }
  .popup-box {
    height: 100%;
  }
  .nut-calendar-content {
    overflow-y: auto;
  }
  ::-webkit-scrollbar {
    display: none;
  }

  // 头部导航
  .nut-calendar-header {
    display: flex;
    flex-direction: column;
    text-align: center;
    padding-top: 1px;
    background-color: $white;

    .calendar-title {
      font-size: $calendar-title-font;
      font-weight: $calendar-title-font-weight;
      line-height: 44px;
    }

    .calendar-curr-month {
      padding: 7px 0;
      line-height: 22px;
      font-size: $calendar-sub-title-font;
    }
    .calendar-top-slot {
      height: $calendar-top-slot-height;
    }

    .calendar-weeks {
      display: flex;
      align-items: center;
      justify-content: space-around;
      height: 36px;
      border-radius: 0px 0px 12px 12px;
      box-shadow: 0px 4px 10px 0px rgba($color: #000000, $alpha: 0.06);

      .calendar-week-item {
        &:first-of-type,
        &:last-of-type {
          color: $calendar-day67-font-color;
        }
      }
    }
  }

  // 月份
  .nut-calendar-content {
    flex: 1;
    width: 100%;
    display: block;
    .calendar-months-panel {
      position: relative;
      width: 100%;
      height: auto;
      display: block;
      box-sizing: border-box;
      .viewArea {
        display: block;
      }
      .calendar-month {
        display: flex;
        flex-direction: column;
        text-align: center;
      }

      div:nth-of-type(2) {
        .calendar-month-title {
          padding-top: 0;
        }
      }

      .calendar-loading-tip {
        height: 50px;
        line-height: 50px;
        text-align: center;
        position: absolute;
        top: -50px;
        left: 0;
        right: 0;
        font-size: $calendar-text-font;
        color: $text-color;
      }

      .calendar-month-title {
        height: 23px;
        line-height: 23px;
        margin: 8px 0;
      }

      .calendar-month-con {
        overflow: hidden;

        .calendar-month-item {
          .calendar-month-day:nth-child(7n + 0),
          .calendar-month-day:nth-child(7n + 1) {
            color: $calendar-day67-font-color;
          }
        }

        .calendar-month-day {
          float: left;
          width: $calendar-day-width;
          height: $calendar-day-height;
          font-weight: $calendar-day-font-weight;
          display: flex;
          align-items: center;
          justify-content: center;
          flex-direction: column;
          position: relative;

          .calendar-curr-tips {
            position: absolute;
            width: 100%;
          }

          .calendar-curr-tip-curr {
            position: absolute;
            bottom: 6px;
            width: 100%;
            font-size: 12px;
            line-height: 14px;
          }
          .calendar-day-tip {
            position: absolute;
            bottom: 6px;
            width: 100%;
            font-size: 12px;
            line-height: 14px;
            color: $calendar-primary-color;
          }
          .calendar-curr-tips-top {
            top: 6px;
          }
          .calendar-curr-tips-bottom {
            bottom: 6px;
          }

          &-active {
            background-color: $calendar-primary-color;
            color: $white !important;
            border-radius: $calendar-day-active-border-radius;

            .calendar-curr-tips {
              visibility: hidden;
            }
            .calendar-curr-tip-curr {
              visibility: hidden;
            }
            .calendar-day-tip {
              color: $white;
            }
          }

          &-disabled {
            color: $calendar-disable-color !important;
          }

          &-choose {
            background-color: $calendar-choose-color;
            color: $calendar-choose-font-color;
          }

          .calendar-day {
            padding: 2px 0;
            font-size: $calendar-day-font;
          }
        }
      }
    }
  }

  // 底部导航
  .nut-calendar-footer {
    display: flex;
    height: 62px;
    width: 100%;
    background-color: $white;

    .calendar-confirm-btn {
      height: 44px;
      width: 100%;
      margin: 10px 18px;
      border-radius: 22px;
      background: $button-primary-background-color;
      color: $white;
      text-align: center;
      line-height: 44px;
    }
  }
}
