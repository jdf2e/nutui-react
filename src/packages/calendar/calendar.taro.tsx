import React, { FunctionComponent } from 'react'
import Popup from '@/packages/popup/index.taro'
import CalendarItem from '@/packages/calendaritem/index.taro'
import Utils from '@/utils/date'
import { useConfig } from '@/packages/configprovider/configprovider.taro'

export interface CalendarProps {
  type?: string
  isAutoBackFill?: boolean
  poppable?: boolean
  visible?: boolean
  title?: string
  defaultValue?: string | string[]
  startDate?: string
  endDate?: string
  onClose?: () => void
  onChoose?: (param: string) => void
}

const defaultProps = {
  type: 'one',
  isAutoBackFill: false,
  poppable: true,
  visible: false,
  title: '日历选择',
  defaultValue: '',
  startDate: Utils.getDay(0),
  endDate: Utils.getDay(365),
  onClose: () => {},
  onChoose: (param: string) => {},
} as CalendarProps

export const Calendar: FunctionComponent<
  Partial<CalendarProps> & React.HTMLAttributes<HTMLDivElement>
> = (props) => {
  const { locale } = useConfig()
  const {
    children,
    poppable,
    visible,
    type,
    isAutoBackFill,
    title,
    defaultValue,
    startDate,
    endDate,
    onClose,
    onChoose,
  } = { ...defaultProps, ...props }

  const close = () => {
    onClose && onClose()
  }

  const update = () => {}
  const choose = (param: string) => {
    close()
    onChoose && onChoose(param)
  }
  const closePopup = () => {
    close()
  }

  return (
    <>
      {poppable ? (
        <Popup
          visible={visible}
          position="bottom"
          round
          duration={0.5}
          closeable
          destroyOnClose
          onClickOverlay={closePopup}
          onClickCloseIcon={closePopup}
          style={{ height: '85vh' }}
        >
          <CalendarItem
            type={type}
            isAutoBackFill={isAutoBackFill}
            poppable={poppable}
            title={locale.calendaritem.title || title}
            defaultValue={defaultValue}
            startDate={startDate}
            endDate={endDate}
            onUpdate={update}
            onClose={close}
            onChoose={choose}
          />
        </Popup>
      ) : (
        <CalendarItem
          type={type}
          isAutoBackFill={isAutoBackFill}
          poppable={poppable}
          title={locale.calendaritem.title}
          defaultValue={defaultValue}
          startDate={startDate}
          endDate={endDate}
          onClose={close}
          onChoose={choose}
        />
      )}
    </>
  )
}

Calendar.defaultProps = defaultProps
Calendar.displayName = 'NutCalendar'
