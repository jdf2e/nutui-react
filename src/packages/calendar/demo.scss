.wrapper {
  display: flex;
  padding: 0 40px;
}
.d_div {
  margin: 0px 5px;
  .d_btn {
    background: #fa3f19;
    color: #fff;
    font-size: 12px;
    padding: 2px 8px;
    border-radius: 4px;
    display: inline-block;
    height: 20px;
  }
}

.info {
  font-size: 12px;
  line-height: 14px;
}
