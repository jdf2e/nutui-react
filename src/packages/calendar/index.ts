import { Calendar } from './calendar'
export * from './calendar'
export default Calendar
