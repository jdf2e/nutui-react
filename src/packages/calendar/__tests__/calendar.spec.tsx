import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import { Calendar } from '../calendar'

interface Day {
  day: string | number
  type: string
}

test('show-title prop', async () => {
  const { container, rerender } = render(
    <Calendar
      visible
      defaultValue="2022-03-18"
      startDate="2022-01-01"
      endDate="2022-12-31"
    />
  )

  const canlendarTitle1 = container.querySelectorAll(
    '.nut-calendar-header .nut-calendar-title'
  )

  expect(canlendarTitle1.length).toBe(1)

  rerender(
    <Calendar
      visible
      showTitle={false}
      defaultValue="2022-03-18"
      startDate="2022-01-01"
      endDate="2022-12-31"
    />
  )

  const canlendarTitle2 = container.querySelectorAll(
    '.nut-calendar-header .nut-calendar-title'
  )

  expect(canlendarTitle2.length).toBe(0)
})

test('show-sub-title prop', async () => {
  const { container, rerender } = render(
    <Calendar
      visible
      showSubTitle
      defaultValue="2022-03-18"
      startDate="2022-01-01"
      endDate="2022-12-31"
    />
  )

  const curMonth1 = container.querySelectorAll('.nut-calendar-sub-title')

  expect(curMonth1.length).toBe(1)

  rerender(
    <Calendar
      visible
      showSubTitle={false}
      defaultValue="2022-03-18"
      startDate="2022-01-01"
      endDate="2022-12-31"
    />
  )

  const curMonth2 = container.querySelectorAll('.nut-calendar-sub-title')

  expect(curMonth2.length).toBe(0)
})

test('show-today prop', async () => {
  const { container } = render(
    <Calendar
      visible
      showToday={false}
      defaultValue="2022-03-18"
      startDate="2022-01-01"
      endDate="2022-12-31"
    />
  )

  const tipCurr = container.querySelectorAll('.nut-calendar-day-info-curr')

  expect(tipCurr.length).toBe(0)
})

test('should render slot correctly', async () => {
  const renderHeaderButtons = () => {
    return <div className="d_div"> 最近七天</div>
  }

  const renderDay = (date: Day) => {
    return <span>custom{date.day}</span>
  }

  const renderDayBottom = (date: Day) => {
    return <span>{Number(date.day) <= 10 ? '上旬' : '下旬'}</span>
  }

  const { container } = render(
    <Calendar
      visible
      defaultValue="2022-03-18"
      startDate="2022-01-01"
      endDate="2022-12-31"
      renderHeaderButtons={renderHeaderButtons}
      renderDay={renderDay}
      renderDayBottom={renderDayBottom}
    />
  )

  const topSlot = container.querySelector(
    '.nut-calendar-header-buttons'
  ) as HTMLElement
  const viewArea = container.querySelector('.viewArea') as HTMLElement
  expect(topSlot.innerHTML).toContain('<div class="d_div"> 最近七天</div>')
  expect(viewArea.innerHTML).toMatchSnapshot()
})

test('select event when click item', () => {
  const onDayClick = jest.fn()
  const { container } = render(
    <Calendar
      visible
      defaultValue="2022-03-18"
      startDate="2022-01-01"
      endDate="2022-12-31"
      onDayClick={onDayClick}
    />
  )

  const calendarMonthDay = container.querySelectorAll('.nut-calendar-day')[15]

  fireEvent.click(calendarMonthDay)
  expect(onDayClick).toBeCalled()
})

test('choose event when click item', async () => {
  const onConfirm = jest.fn()
  const { container } = render(
    <Calendar
      visible
      defaultValue="2022-03-18"
      startDate="2022-01-01"
      endDate="2022-12-31"
      onConfirm={onConfirm}
    />
  )

  const calendarConfirmBtn = container.querySelectorAll(
    '.calendar-confirm-btn'
  )[0]

  fireEvent.click(calendarConfirmBtn)
  expect(onConfirm).toBeCalled()
})
