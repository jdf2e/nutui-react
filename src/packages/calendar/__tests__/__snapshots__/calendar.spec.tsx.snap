// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render slot correctly 1`] = `"<div class="nut-calendar-month"><div class="nut-calendar-month-title">2022年02月</div><div class="nut-calendar-days"><div class="nut-calendar-day nut-calendar-day-disabled"><div class="nut-calendar-day-day"><span>custom30</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day nut-calendar-day-disabled"><div class="nut-calendar-day-day"><span>custom31</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom1</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom2</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom3</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom4</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom5</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom6</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom7</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom8</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom9</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom10</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom11</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom12</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom13</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom14</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom15</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom16</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom17</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom18</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom19</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom20</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom21</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom22</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom23</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom24</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom25</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom26</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom27</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom28</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div></div></div><div class="nut-calendar-month"><div class="nut-calendar-month-title">2022年03月</div><div class="nut-calendar-days"><div class="nut-calendar-day nut-calendar-day-disabled"><div class="nut-calendar-day-day"><span>custom27</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day nut-calendar-day-disabled"><div class="nut-calendar-day-day"><span>custom28</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom1</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom2</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom3</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom4</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom5</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom6</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom7</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom8</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom9</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom10</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom11</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom12</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom13</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom14</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom15</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom16</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom17</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day nut-calendar-day-active"><div class="nut-calendar-day-day"><span>custom18</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom19</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom20</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom21</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom22</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom23</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom24</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom25</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom26</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom27</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom28</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom29</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom30</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom31</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div></div></div><div class="nut-calendar-month"><div class="nut-calendar-month-title">2022年04月</div><div class="nut-calendar-days"><div class="nut-calendar-day nut-calendar-day-disabled"><div class="nut-calendar-day-day"><span>custom27</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day nut-calendar-day-disabled"><div class="nut-calendar-day-day"><span>custom28</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day nut-calendar-day-disabled"><div class="nut-calendar-day-day"><span>custom29</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day nut-calendar-day-disabled"><div class="nut-calendar-day-day"><span>custom30</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day nut-calendar-day-disabled"><div class="nut-calendar-day-day"><span>custom31</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom1</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom2</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom3</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom4</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom5</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom6</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom7</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom8</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom9</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom10</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom11</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom12</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom13</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom14</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom15</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom16</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom17</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom18</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom19</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom20</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom21</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom22</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom23</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom24</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom25</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom26</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom27</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom28</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom29</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom30</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div></div></div><div class="nut-calendar-month"><div class="nut-calendar-month-title">2022年05月</div><div class="nut-calendar-days"><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom1</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom2</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom3</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom4</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom5</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom6</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom7</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom8</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom9</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom10</span></div><div class="nut-calendar-day-info-bottom"><span>上旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom11</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom12</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom13</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom14</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom15</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom16</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom17</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom18</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom19</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom20</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom21</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom22</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom23</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom24</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom25</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom26</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom27</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom28</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom29</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom30</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div><div class="nut-calendar-day "><div class="nut-calendar-day-day"><span>custom31</span></div><div class="nut-calendar-day-info-bottom"><span>下旬</span></div></div></div></div>"`;
